jobbergate-api/jobbergate_api/apps/job_script_templates/routers.py

```diff
@@ -381,6 +458,49 @@ async def job_script_workflow_upload_file(
     return await secure_services.file.workflow.upsert(**upsert_kwargs)
 
 
+@router.put(
+    "/{id_or_identifier}/upload/workflow",
+    status_code=status.HTTP_200_OK,
+    description="Endpoint to upload a file to a job script template by id or identifier",
+    response_model=WorkflowFileDetailedView,
+)
+async def job_script_workflow_upload_file(
+    id_or_identifier: str = Path(),
+    runtime_config: RunTimeConfig | None = Body(
+        None, description="Runtime configuration is optional when the workflow file already exists"
+    ),
+    upload_file: UploadFile = File(..., description="File to upload"),
+    secure_services: SecureService = Depends(
+        secure_services(Permissions.ADMIN, Permissions.JOB_TEMPLATES_CREATE, ensure_email=True)
+    ),
+):
+    """Upload a file to a job script workflow by id or identifier."""
+    return await _upsert_workflow_file(id_or_identifier, runtime_config, upload_file, secure_services)
+
+
+@router.put(
+    "/{id_or_identifier}/upload-by-url/workflow",
+    status_code=status.HTTP_200_OK,
+    description="Endpoint to upload a file to a job script template by id or identifier from a provided URL.",
+    response_model=WorkflowFileDetailedView,
+)
+async def job_script_upload_file_by_url(
+    id_or_identifier: str = Path(),
+    runtime_config: RunTimeConfig | None = Body(
+        None, description="Runtime configuration is optional when the workflow file already exists"
+    ),
+    file_url: AnyHttpUrl = Query(..., description="URL of the file to upload"),
```

Comment:
Both here and on the other endpoint, I think it should support s3 URIs, e.g. `s3://<bucket name>/<file>`. Then, in [here](https://github.com/omnivector-solutions/jobbergate/blob/49c4533175c71837f53a29e8ff9343de5371aabc/jobbergate-api/jobbergate_api/apps/services.py#L566), it would require to identify the schema (s3 or http/https) and apply the proper logic.

Reply:
Added support for s3 in commit a3ebd51.

Reply:
shouldn't this endpoint also accept `AnyUrl` instead of `AnyHttpUrl`? The workflow file also can be from s3, right?

Reply:
Damn, I missed that. Thanks.

Reply:
Addressed in 370c6dd.